packages/modules/vehicles/polestar/api.py

```diff
@@ -0,0 +1,106 @@
+import logging
+import requests
+import time
+from modules.vehicles.polestar.auth import PolestarAuth
+from modules.common.component_state import CarState
+
+
+log = logging.getLogger(__name__)
+
+
+class PolestarApi:
+
+    def __init__(self, username: str, password: str, vin: str) -> None:
+        self.auth = PolestarAuth(username, password, vin)
+        self.vin = vin
+        self.client_session = requests.session()
+
+    def query_params(self, params: dict, url='https://pc-api.polestar.com/eu-north-1/mystar-v2/') -> dict or None:
+        access_token = self.auth.get_auth_token()
+        if access_token is None:
+            log.error("query_params:not yet authenticated")
+            return None
+
+        headers = {
+            "Content-Type": "application/json",
+            "authorization": f"Bearer {self.auth.access_token}"
+        }
+
+        log.info("query_params:%s", params['query'])
+        try:
+            result = self.client_session.get(url=url, params=params, headers=headers)
+        except requests.RequestException as e:
+            log.error("query_params:http error:%s", e)
+            return None
+
+        if result.status_code == 401:
+            self.auth.delete_token()
+            log.error("query_params error: get response %d: unauthorized Exception", result.status_code)
+            return None
+        if result.status_code != 200:
+            log.error("query_params error: get response %d: %s", result.status_code, result.text)
+            return None
+
+        result_data = result.json()
+        if result_data.get('errors'):
+            error_message = result_data['errors'][0]['message']
+            log.error("query_params error: %s", error_message)
+            return None
```

Comment:
```suggestion
            self.auth.delete_token()
           raise Exception("query_params error: get response %d: unauthorized Exception", result.status_code)
        if result.status_code != 200:
            raise Exception("query_params error: get response %d: %s", result.status_code, result.text)

        result_data = result.json()
        if result_data.get('errors'):
            error_message = result_data['errors'][0]['message']
            raise Exception("query_params error: %s", error_message)
```
Hier macht es Sinn, die weitere Verarbeitung abzubrechen und eine Exception zu heben.

Reply:
Einverstanden